### ISSUE TITLE
[Bug]  Training GDN with flash-linear-attention hangs after 8k–12k steps on a single H200

### ISSUE THREAD

**Issue by Sky-Trigger** [fla-org/flash-linear-attention#947]
### Checklist

- [x] I have checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md) and existing issues for similar problems
- [x] Please report this bug in English to ensure wider understanding and support

### Describe the Bug

Hi, I encountered a reproducible hang when training a GDN model with flash-linear-attention on a single NVIDIA H200 GPU.

The training does not crash or raise an error. Instead, it always gets stuck after running for a while, usually between 8,000 and 12,000 training steps. Once it reaches the problematic step, the process stops progressing and never enters the next step. Because of this, I have to manually restart the training process every time.

### Steps to Reproduce the Bug

Model: GDN
Attention/kernel: flash-linear-attention
Hardware: 1 × NVIDIA H200
Training behavior: reproducibly hangs around 8,000–12,000 steps
Failure type: hang / deadlock / no progress
Error message: no explicit error message

### Expected Behavior

Training should continue normally after 8k–12k steps without hanging.

### Environment Information

1. Torch: 2.11.0
2. Triton: 3.6.0

**Comment by Sky-Trigger** [comment]
Actually, I ran multiple additional tests and found that the training does not only hang around the 8k–12k step range. Instead, GDN tends to hang after training for a certain period of time. Each time this happens, I have to manually stop the process and resume training from the previously saved checkpoint.

I am using the standard 340M Gated DeltaNet config from the repository. The exact training parameters are as follows:
MODEL=gated_deltanet-340M
cd /data/8T/wangmb/KVC/Linear/flame && NNODE=1 NGPU=1 LOG_RANK=0 bash train.sh \
    --job.config_file flame/models/fla.toml \
    --job.dump_folder exp/$MODEL \
    --model.config configs/${MODEL/-/_}.json \
    --model.tokenizer_path /data/8T/wangmb/KVC/Linear/checkpoints/delta_net-1.3B-100B \
    --optimizer.name AdamW \
    --optimizer.lr 3e-4 \
    --lr_scheduler.warmup_steps 1024 \
    --lr_scheduler.decay_type cosine \
    --training.batch_size 8 \
    --training.seq_len 2048 \
    --training.gradient_accumulation_steps 4 \
    --training.steps 40960 \
    --training.max_norm 1.0 \
    --training.dataset HuggingFaceFW/fineweb-edu \
    --training.dataset_name sample-100BT \
    --training.dataset_split train \
    --training.streaming \
    --training.num_workers 4 \
    --training.seed 42 \
    --training.disable_loss_parallel \
    --checkpoint.interval 2048 \
    --checkpoint.load_step -1 \
    --metrics.log_freq 1
